model/ring_request.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+	"net/url"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// CreateRingRequest specifies the parameters for a new ring.
+type CreateRingRequest struct {
+	Name              string `json:"name,omitempty"`
+	Priority          int    `json:"priority,omitempty"`
+	InstallationGroup string `json:"installationGroup,omitempty"`
+	SoakTime          int    `json:"soakTime,omitempty"`
+	Image             string `json:"image,omitempty"`
+	Version           string `json:"version,omitempty"`
+	APISecurityLock   bool   `json:"api-security-lock,omitempty"`
```

Comment:
I would suggest keeping it consistent if we use camel case for others.
```suggestion
	APISecurityLock   bool   `json:"apiSecurityLock,omitempty"`
```

Reply:
Thanks for the suggestion, fixed